overlord/patch/export_test.go

```diff
@@ -16,3 +16,8 @@ func MockReadInfo(f func(name string, si *snap.SideInfo) (*snap.Info, error)) (r
 	readInfo = f
 	return func() { readInfo = old }
 }
+
+var (
+	ApplyOne = applyOne
+	Patch1   = patch1
```

Comment:
Instead of exporting these internals, I suggest implementing a `MockLevel` function here in `export_test.go`.

Then, the test would simply do MockLevel(1) (with the usual restore defer) and pass the state into the real Apply method. This is both nicer as it requires less knowledge of the internals, and more complete.